core-web/libs/utils/src/lib/dot-utils.spec.ts

```diff
@@ -367,4 +367,17 @@ describe('Dot Utils', () => {
             );
         });
     });
+
+    describe('hasValidValue', () => {
+        it('should return true when value is not empty string, null, or undefined', () => {
+            expect(hasValidValue('test')).toEqual(true);
+        });
+
+        it('should return false when value is empty string, null, or undefined', () => {
+            expect(hasValidValue('')).toEqual(false);
+            expect(hasValidValue('   ')).toEqual(false);
+            expect(hasValidValue(null)).toEqual(false);
+            expect(hasValidValue(undefined)).toEqual(false);
```

Comment:
We should create a test case for each expect

We can add multiple expects when we want to test a whole flow of data, in this case, we need one for each expect.